lib/core/src/recover/recoverer.rs

```diff
@@ -768,6 +771,19 @@ impl Recoverer {
                 false => btc_last_outgoing_tx_id,
             };
 
+            // Verify amount - if not expected, ignore any potential server lockup tx
+            let payer_amount_sat = receive_chain_swap_immutable_data_by_swap_id
+                .get(&swap_id)
+                .map(|imm| imm.payer_amount_sat)
+                .ok_or_else(|| {
+                    anyhow!("Initial payer amount not found for Onchain Receive Swap {swap_id}")
+                })?;
```

Comment:
We're going to bail on the whole fn here, needs to handle no value more gracefully

Reply:
This is the strategy currently used in other places in the recoverer. These errors should never happen, but we can consider switching to ignoring just the "corrupted" swap. If so, I propose we do it in a different PR.

Reply:
@dangeross I agree with Daniel, the error is on the failure to retrieve from the hash map and already handled like this before this line: https://github.com/breez/breez-sdk-liquid/pull/667/files#diff-28794968ead61e86665263d494740e1a8fbe62fcfed42ed568bcf77fd51ca31dR712